feat: add C ndarray interface and refactor implementation for `stats/base/dvariancetk`

aayush0325 opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C ndarray interface for `stats/base/dvariancetk`
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aayush0325 committed:
feat: add C ndarray interface and refactor implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

DhruvArvindSingh commented:
@aayush0325 I have a small doubt.... Why have you used CBLAS_INT instead of int64_t in the main.c file?

aayush0325 commented:
> @aayush0325 I have a small doubt.... Why have you used CBLAS_INT instead of int64_t in the main.c file?

I believe it is a convention thing, if you'll see [stdlib/blas/base/shared.h](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/base/shared/include/stdlib/blas/base/shared/cblas.h) you'll notice that `CBLAS_INT` is a wrapper over `int32_t` and `int64_t` based on specific conditions during compilation. rest i believe @kgryte can give a better explanation than i can.

aayush0325 committed:
feat: missing changes in main.c

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented:
@DhruvArvindSingh `CBLAS_INT` stems from `@stdlib/blas/base/*`. This value can be set at compilation to create either 32- or 64-bit interfaces, depending on hardware support, etc. This follows BLAS, which supports something similar. (aside: historically, BLAS was only 32-bit. Only recently was 64-bit support added.) While most modern platforms have 64-bit support, that is not true of all systems, particularly embedded systems with limited memory. By supporting `API_SUFFIX` and `CBLAS_INT`, we ensure consistent conventions across our strided array interfaces, not just in BLAS.

kgryte reviewed (approved):
LGTM
